src/differential_geometry.jl

```diff
@@ -22,27 +72,133 @@ function Lift(X::Function, dependences::DataType...)::HamiltonianLift
 end
 
 # ---------------------------------------------------------------------------
-# Lie derivative of a scalar function along a vector field: L_X(f) = X⋅f
+# Lie derivative of a scalar function along a vector field or a function: L_X(f) = X⋅f
 
 # (postulat)
 # (X⋅f)(x)    = f'(x)⋅X(x)
 # (X⋅f)(t, x) = ∂ₓf(t, x)⋅X(t, x)
+# (g⋅f)(x) = f'(x)⋅G(x) with G the vector of g
```

Comment:
@BaptisteCbl Was ist das?

Reply:
That's what we wanted (at least at the time) for our functions. I don't really know if there's any point in keeping it somewhere or not, but since you're asking, I suppose it should be deleted.

Reply:
@BaptisteCbl Late answer on this: not sure about the notion of Lie derivative of a function along *a function* (Lie derivative are *wrt.* vector fields). It is true that in the symplectic case there is a vector field canonically associated with a function (its symplectic gradient).

Reply:
Hello, I'm not sure I understand the answer well but. It is the Lie derivative of a function (Julia type : Function) along a vector field (Julia type : Function). The "function" only refers to the Julia type and not the mathematical object (since we uses a julia Function to define a vector field we might want to use the function directly instead of the object VectorField). 
For example : 
```julia
julia> φ = x -> [x[2], -x[1]]
julia> f = x -> x[1]^2 + x[2]^2
julia> Lie(φ,f)([1, 2])
0
```
I wrote it in a confusing way perhaps it was unclear to me at the time.